text/3-entrees/fixed-point.md

```diff
@@ -0,0 +1,327 @@
+	Ok(())
+}
+```
+
+The code of this extrinsic largely speaks for itself. One thing to take particular note of is that we _don't_ check for overflow on the multiplication. If you've read many of the recipes you know that a Substrate runtime must never panic, and a developer must be extremely diligent in always checking for and gracefully handling error conditions. Because `Permill` only holds values between 0 and 1, we know that their product will always be in that same range. Thus it is impossible to overflow or saturate. So we can happily use `saturating_mul` and move on.
+
+### Substrate-fixed Accumulator
+
+[Substrate-fixed](https://github.com/encointer/substrate-fixed/) takes a more traditional approach in that their types represent numbers with both whole _and_ fractional parts. For this fractional part, we'll use the `U16F16` type. This type contains an unsigned number (indicated by the `U` at the beginning) and has 32 _total_ bits of precision - 16 for the integer part, and 16 for the fractional part. There are several other types provided that follow the same naming convention. Some examples include `U32F32` and `I32F32` where the `I` indicates a signed number, just like in Rust primitive types.
+
+As in the `Permill` example, we begin by declaring the storage item. With substrate-fixed, there is not a `one` function, but there is a `from_num` function that we use to set the storage item's default value. This `from_num` method and it's counterpart `to-num` are your primary ways of converting between substrate-fixed types and Rust primitive types. If your use case does a lot of fixed-point arithmetic, like ours does, it is advisable to keep your data in substrate-fixed types.
+
+> We're able to use `U16F16` as a storage item type because it, and the other substrate-fixed types, implement the parity scale codec.
+
+```rust, ignore
+decl_storage! {
+	trait Store for Module<T: Trait> as Example {
+		// --snip--
+
+		/// Substrate-fixed accumulator, value starts at 1 (multiplicative identity)
+		FixedAccumulator get(fn fixed_value): U16F16 = U16F16::from_num(1);
+	}
+}
+```
+
+Next we implement the extrinsic that allows users to update the accumulator by multiplying in a new value.
+
+```rust, ignore
+fn update_fixed(origin, new_factor: U16F16) -> DispatchResult {
+	let _ = ensure_signed(origin)?;
+
+	let old_accumulated = Self::fixed_value();
+
+	// Multiply, handling overflow
+	let new_product = old_accumulated.checked_mul(new_factor)
+		.ok_or(Error::<T>::Overflow)?;
+
+	// Write the new value to storage
+	FixedAccumulator::put(new_product);
+
+	// Emit event
+	Self::deposit_event(Event::FixedUpdated(new_factor, new_product));
+	Ok(())
+}
+```
+
+This extrinsic is quite similar to the `Permill` ersion with one notable difference. Because `U16F16` handles numbers greater than one, overflow is possible, and we need to handle it. The error handling here is straightforward, the important part is just that you remember to do it.
```

Comment:
Your example is super simple and allows to compare the different approaches. However, if you don't mind, I'd mention the possibility to use additional math functions like *log*, *sqrt*, *exp*, *sin*, *pow*. Because that's what you can't do with the other options.

Reply:
I added a note mentioning that we'll see cooler features of substrate-fixed later on.